src/System.Text.Encodings.Web.Utf8/UrlEncoder.cs

```diff
@@ -297,79 +297,30 @@ private static int UnescapePercentEncoding(ref int scan, Span<byte> buffer)
                 return -1;
             }
 
-            var probe = scan;
-
-            var value1 = ReadHex(ref probe, buffer);
-            if (value1 == -1)
+            buffer = buffer.Slice(scan);
+            if (buffer.Length < 2)
             {
                 return -1;
             }
 
-            var value2 = ReadHex(ref probe, buffer);
-            if (value2 == -1)
+            if (!Utf8Parser.TryParse(buffer.Slice(0, 2), out byte value, out var bytesConsumed, 'X'))
             {
                 return -1;
             }
-
-            if (SkipUnescape(value1, value2))
+            
+            if (bytesConsumed != 2)
```

Comment:
When will `bytesConsumed != 2` here if TryParse returned true?

Reply:
It could be a single digit hex (i.e. "%FZ")

Reply:
Ah yes, good point.

Reply:
Is there a unit test for the "%FZ" condition? If not, we should add one. Overall LGTM.

Reply:
@GrabYourPitchforks there wasn't. I added one.